test/extensions/filters/http/rbac/rbac_filter_test.cc

```diff
@@ -68,22 +72,42 @@ class RoleBasedAccessControlFilterTest : public testing::Test {
     ON_CALL(connection_, requestedServerName()).WillByDefault(Return(requested_server_name_));
   }
 
+  void checkAccessLogMetadata(bool expected) {
+    auto filter_meta = req_info_.dynamicMetadata().filter_metadata().at(
+        Filters::Common::RBAC::DynamicMetadataKeysSingleton::get().CommonNamespace);
+    EXPECT_EQ(expected,
+              filter_meta.fields()
+                  .at(Filters::Common::RBAC::DynamicMetadataKeysSingleton::get().AccessLogKey)
+                  .bool_value());
+  }
+
   void setMetadata() {
     ON_CALL(req_info_, setDynamicMetadata(HttpFilterNames::get().Rbac, _))
         .WillByDefault(Invoke([this](const std::string&, const ProtobufWkt::Struct& obj) {
           req_info_.metadata_.mutable_filter_metadata()->insert(
               Protobuf::MapPair<std::string, ProtobufWkt::Struct>(HttpFilterNames::get().Rbac,
                                                                   obj));
         }));
+
+    ON_CALL(req_info_,
+            setDynamicMetadata(
+                Filters::Common::RBAC::DynamicMetadataKeysSingleton::get().CommonNamespace, _))
+        .WillByDefault(Invoke([this](const std::string&, const ProtobufWkt::Struct& obj) {
+          req_info_.metadata_.mutable_filter_metadata()->insert(
+              Protobuf::MapPair<std::string, ProtobufWkt::Struct>(
+                  Filters::Common::RBAC::DynamicMetadataKeysSingleton::get().CommonNamespace, obj));
+        }));
   }
 
   NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks_;
   NiceMock<Network::MockConnection> connection_{};
   NiceMock<Envoy::StreamInfo::MockStreamInfo> req_info_;
   Stats::IsolatedStoreImpl store_;
   RoleBasedAccessControlFilterConfigSharedPtr config_;
-
   RoleBasedAccessControlFilter filter_;
+  RoleBasedAccessControlFilterConfigSharedPtr log_config_;
```

Comment:
no, let's use only one filter_ and config_ otherwise it gets really complicated of which one is being tested in a test case.

You can have a custom setUp(envoy::config::rbac::v3::RBAC::Action action) that creates a new config/filter and assign to the config_/filter_ in the beginning of each test case, it also combine the SetUp().

Similar to what you are doing in the network/rbac/filter_test.cc

Reply:
This makes more verbose to check log and non-log actions in the same test, which I do a couple times, but for clarity I can change it.

Reply:
why more verbose? they are testing different things, for a non-LOG action, we also need to make sure it doesn't write to the metadata.

Reply:
I'm testing both configs in some tests, see below comment.